packages/components/src/__actions__/Button/v3/types.ts

```diff
@@ -0,0 +1,21 @@
+export type ButtonVariants = "primary" | "secondary" | "tertiary"
+
+export type ButtonSizes = "small" | "medium" | "large"
+
+export type PendingPropsUndefined = {
+  isPending?: undefined
+  pendingLabel?: never
+  isPendingLabelHidden?: never
```

Comment:
To check: Last time I wrote a discriminated union for props, the description needed to be defined in all places to show in the docs. Check whether this still needs to be done.

Reply:
They come through in the docs, but probably worth having since we export the type
![Screenshot 2024-11-07 at 11 36 28 AM](https://github.com/user-attachments/assets/812a382d-3ea5-4f79-9f0d-142efd058488)

Reply:
Little bit of double up but I've amended this for safety 🦺